### PR TITLE
Simplify dependencies

### LINKED ISSUE #539: ncov conda environment failure on creation
**Current Behavior**
The conda environment configuration for ncov fails to create a working environment. An error indicating that Cython is missing is reported (see end of issue). If Cython is added into the config prior to environment creation, the environment fails due to a Python 3.7 requirement (not shown). 

**Expected behavior**
It should be possible to setup a ncov environment by following the setup documentation.

**How to reproduce**
Steps to reproduce the current behavior:
1. Go to "https://nextstrain.github.io/ncov/setup"
2. Copy and paste the first two code cells into a terminal

**Possible solution**
Following failure, a manual install of the missing pip dependencies seemed to work. It's not obvious why that works when conda didn't.

**Your environment: if running Nextstrain locally**
 - Operating system: Centos 6.6
 - Conda 4.9.2

**Conda error**

```bash
$ curl http://data.nextstrain.org/nextstrain.yml --compressed -o nextstrain.yml
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   300  100   300    0     0   2930      0 --:--:-- --:--:-- --:--:--  6976
$ md5sum nextstrain.yml
0b615193021c854cae13c910b5d4417c  nextstrain.yml
$ conda env create -f nextstrain.yml

Collecting package metadata (repodata.json): done
Solving environment: done
Preparing transaction: done
Verifying transaction: done
Executing transaction: done
Installing pip dependencies: \ Ran pip subprocess with arguments:
['/home/mcdonadt/miniconda3/envs/nextstrain/bin/python', '-m', 'pip', 'install', '-U', '-r', '/home/mcdonadt/condaenv.jju_bmce.requirements.txt']
Pip subprocess output:
Downloading/unpacking awscli (from -r /home/mcdonadt/condaenv.jju_bmce.requirements.txt (line 1))
  Running setup.py egg_info for package awscli
    /tmp/pip_build_dtmcdonald/awscli/setup.py:11: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately.

Downloading/unpacking nextstrain-augur (from -r /home/mcdonadt/condaenv.jju_bmce.requirements.txt (line 2))
  Running setup.py egg_info for package nextstrain-augur
    /tmp/pip_build_dtmcdonald/nextstrain-augur/setup.py:11: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately.
      This may be due to an out of date pip.
Downloading/unpacking nextstrain-cli (from -r /home/mcdonadt/condaenv.jju_bmce.requirements.txt (line 3))
  Running setup.py egg_info for package nextstrain-cli
    /tmp/pip_build_dtmcdonald/nextstrain-cli/setup.py:11: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately.

Downloading/unpacking rethinkdb==2.3.0.post6 (from -r /home/mcdonadt/condaenv.jju_bmce.requirements.txt (line 4))
  Running setup.py egg_info for package rethinkdb
    /tmp/pip_build_dtmcdonald/rethinkdb/setup.py:11: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately.
      import os, setuptools, sys
Downloading/unpacking snakemake==5.26.1 (from -r /home/mcdonadt/condaenv.jju_bmce.requirements.txt (line 5))
  Running setup.py egg_info for package snakemake
    /tmp/pip_build_dtmcdonald/snakemake/setup.py:11: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately.
      import versioneer
Downloading/unpacking botocore==1.19.59 (from awscli->-r /home/mcdonadt/condaenv.jju_bmce.requirements.txt (line 1))
  Running setup.py egg_info for package botocore
    /tmp/pip_build_dtmcdonald/botocore/setup.py:11: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately.

    warning: no files found matching 'botocore/vendored/requests/cacert.pem'
    no previously-included directories found matching 'docs/build'
Downloading/unpacking docutils<0.16,>=0.10 (from awscli->-r /home/mcdonadt/condaenv.jju_bmce.requirements.txt (line 1))
  Running setup.py egg_info for package docutils
    /tmp/pip_build_dtmcdonald/docutils/setup.py:11: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately.
      from distutils.command.build import build
    warning: no files found matching 'MANIFEST'
    warning: no previously-included files matching '*.pyc' found under directory '*'
    warning: no previously-included files matching '*~' found under directory '*'
    warning: no previously-included files matching '.DS_Store' found under directory '*'
Downloading/unpacking s3transfer<0.4.0,>=0.3.0 (from awscli->-r /home/mcdonadt/condaenv.jju_bmce.requirements.txt (line 1))
  Running setup.py egg_info for package s3transfer
    /tmp/pip_build_dtmcdonald/s3transfer/setup.py:11: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately.

Downloading/unpacking PyYAML<5.4,>=3.10 (from awscli->-r /home/mcdonadt/condaenv.jju_bmce.requirements.txt (line 1))
  Running setup.py egg_info for package PyYAML
    /tmp/pip_build_dtmcdonald/PyYAML/setup.py:11: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately.
      support, capable extension API, and sensible error messages.  PyYAML
Downloading/unpacking colorama<0.4.4,>=0.2.5 (from awscli->-r /home/mcdonadt/condaenv.jju_bmce.requirements.txt (line 1))
  Downloading colorama-0.4.3.tar.gz
  Running setup.py egg_info for package colorama
    /tmp/pip_build_dtmcdonald/colorama/setup.py:11: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately.
      from distutils.core import setup
Downloading/unpacking rsa<=4.5.0,>=3.1.2 (from awscli->-r /home/mcdonadt/condaenv.jju_bmce.requirements.txt (line 1))
  Running setup.py egg_info for package rsa
    /tmp/pip_build_dtmcdonald/rsa/setup.py:11: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately.
      #  Unless required by applicable law or agreed to in writing, software
Downloading/unpacking bcbio-gff==0.6.*,>=0.6.0 (from nextstrain-augur->-r /home/mcdonadt/condaenv.jju_bmce.requirements.txt (line 2))
  Downloading bcbio-gff-0.6.6.tar.gz
  Running setup.py egg_info for package bcbio-gff
    /tmp/pip_build_dtmcdonald/bcbio-gff/setup.py:11: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately.
      exec(line.strip())
Downloading/unpacking biopython<=1.76,>=1.67 (from nextstrain-augur->-r /home/mcdonadt/condaenv.jju_bmce.requirements.txt (line 2))
  Running setup.py egg_info for package biopython
    /tmp/pip_build_dtmcdonald/biopython/setup.py:11: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately.
      the PyPI archive with::
    warning: no previously-included files found matching 'Tests/Gck/DGVC_GCK.zip'
    warning: no previously-included files matching '*.pyc' found anywhere in distribution
    warning: no previously-included files matching '*.pyo' found anywhere in distribution
    warning: no previously-included files matching '*.py{}' found anywhere in distribution
    warning: no previously-included files matching '*.py-e' found anywhere in distribution
    warning: no previously-included files found matching 'Bio/Align/substitution_matrices/data/README.txt'
Downloading/unpacking jsonschema==3.*,>=3.0.0 (from nextstrain-augur->-r /home/mcdonadt/condaenv.jju_bmce.requirements.txt (line 2))
  Running setup.py egg_info for package jsonschema
    /tmp/pip_build_dtmcdonald/jsonschema/setup.py:11: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately.
Downloading/unpacking packaging>=19.2 (from nextstrain-augur->-r /home/mcdonadt/condaenv.jju_bmce.requirements.txt (line 2))
  Running setup.py egg_info for package packaging
    /tmp/pip_build_dtmcdonald/packaging/setup.py:11: PkgResourcesDeprecationWarning: Parameters to load are deprecated.  Call .resolve and .require separately.
      # setuptools and distutils with a single setup.py, in this specific instance
    warning: no previously-included files found matching 'noxfile.py'
    warning: no previously-included files found matching '.readthedocs.yml'
    warning: no previously-included files found matching '.travis.yml'
    warning: no previously-included files found matching 'dev-requirements.txt'
    warning: no previously-included files found matching 'tests/build-hello-world.sh'
    warning: no previously-included files found matching 'tests/hello-world.c'
    no previously-included directories found matching 'docs/_build'
    no previously-included directories found matching 'tasks'
Downloading/unpacking pandas==1.*,>=1.0.0 from https://files.pythonhosted.org/packages/11/1c/b0bc154996617eae877ff267fcf84e55e6c6808dbade0da206f0419dd483/pandas-1.2.1.tar.gz#sha256=5527c5475d955c0bc9689c56865aaa2a7b13c504d6c44f0aadbf57b565af5ebd (from nextstrain-augur->-r /home/mcdonadt/condaenv.jju_bmce.requirements.txt (line 2))
  Running setup.py egg_info for package pandas
    Traceback (most recent call last):
      File "<string>", line 16, in <module>
      File "/tmp/pip_build_dtmcdonald/pandas/setup.py", line 761, in <module>
        setup_package()
      File "/tmp/pip_build_dtmcdonald/pandas/setup.py", line 731, in setup_package
        ext_modules=maybe_cythonize(extensions, compiler_directives=directives),
      File "/tmp/pip_build_dtmcdonald/pandas/setup.py", line 496, in maybe_cythonize
        raise RuntimeError("Cannot cythonize without Cython installed.")
    RuntimeError: Cannot cythonize without Cython installed.
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):

  File "<string>", line 16, in <module>

  File "/tmp/pip_build_dtmcdonald/pandas/setup.py", line 761, in <module>

    setup_package()

  File "/tmp/pip_build_dtmcdonald/pandas/setup.py", line 731, in setup_package

    ext_modules=maybe_cythonize(extensions, compiler_directives=directives),

  File "/tmp/pip_build_dtmcdonald/pandas/setup.py", line 496, in maybe_cythonize

    raise RuntimeError("Cannot cythonize without Cython installed.")

RuntimeError: Cannot cythonize without Cython installed.

----------------------------------------
Cleaning up...
Command python setup.py egg_info failed with error code 1 in /tmp/pip_build_dtmcdonald/pandas
Storing complete log in /tmp/tmpmk6a8jit

Pip subprocess error:

failed

CondaEnvException: Pip failed
```

### PR DESCRIPTION
**Pull request by huddlej**
Install Augur from [its Bioconda package](https://bioconda.github.io/recipes/augur/README.html) which defines most dependencies required for the Nextstrain environment. This change allows us to remove most dependencies from the Nextstrain environment file and avoid mismatches in dependency versions across environments.

Similarly, this PR also allows Python >=3.6 to be installed instead of pinning to 3.6 specifically. This change should help us avoid breaking environments when dependencies no longer support 3.6 (for example, [when pandas no longer support Python 3.6](https://github.com/nextstrain/ncov/issues/539)).

This PR removes opencv since it is no longer required by any of our workflows and adds substantial complexity to the Nextstrain environment.

Finally, this PR updates the README instructions to recommend Mamba as an alternative to the default conda installer. Other references to our conda environment file should follow the same pattern, so users get the fastest and most accurate environments possible.

Fixes https://github.com/nextstrain/ncov/issues/539

### CODE REVIEW

File: README.md
Comment (tsibley) [inline README.md:22]: I'm not sure about this change, as it feels to me like it introduces another place for confusion in the process. For example, first you have to use Conda to install Mamba. Then you use Mamba to install Nextstrain. But _then_ you go back to Conda again to activate Nextstrain. I understand the nuances here, but introducing Mamba seems it might add confusion for many folks.

How much faster is Mamba for this install? For the typical person, they might only create the environment once or twice. Is it worth it?

(and anyone who already understands the nuances because they use Mamba already, will know that they can use `mamba env create` instead of `conda env create` even if the instructions don't say so.)

Comment (huddlej) [reply README.md:22]: I see what you're saying. It's hard to benchmark the time savings without deleting miniconda and starting from scratch under both scenarios and even then I'm not sure I can consistently recreate the issues I've had in the past with conda-forge. I think the bind we're in is:

1. Bioconda expects conda-forge to be an available channel for all non-biology-related dependencies (and Snakemake relies on this extra channel, too)
2. Adding conda-forge as a channel can be installations either painfully slow or [completely broken](https://github.com/conda/conda/issues/9905) or [both](https://github.com/nextstrain/augur/pull/411)

[We removed conda-forge from this environment](https://github.com/nextstrain/conda/commit/0ad32bc19a6a3e91d812bfbfd1aeeee7f63f05ec) for the last reason.

In practice, Mamba has solved this issue as an independent tool, but those solutions haven't made their way back into Conda and it's not clear if they ever will. This is why [Snakemake recommends Mamba for its own installation](https://snakemake.readthedocs.io/en/stable/getting_started/installation.html#installation-via-conda); it's frustrating to have to work around the package manager itself, but the benefit of being able to consistently install your packages is worth the added complexity of yet another executable.

In terms of minimizing conceptual dependencies for users who get to this README, I don't see `mamba` as more complicated than `curl` or `conda`. My guess is that folks are either going to copy and paste (hoping for the best) or, if they know the nuances, they will copy and paste knowing that it is the better solution for them.

Maybe as an alternative to my current proposal, we could reword the instructions to align with [the current Augur installation docs](https://docs.nextstrain.org/projects/augur/en/stable/installation/installation.html#using-conda)? These suggest Mamba as an alternative for people who want to try it instead of making it the required path.

Comment (huddlej) [reply README.md:22]: db07cda revises the README to make Mamba an alternate path to installation while retaining the current full-Conda approach.

Comment (fanninpm) [reply README.md:22]: Re: conda-forge, see the discussion in bioconda/bioconda-utils#330

Comment (huddlej) [reply README.md:22]: Thank you, @fanninpm! That is very helpful to know that our original channel order may have caused our slow/broken installations in the past.

**Review by tsibley — APPROVED** [review]
Thanks, @huddlej! This looks good to me.

Thoughts on installing `nextstrain-cli` from Bioconda instead of PyPI now that there's a package available?

File: README.md
Comment (tsibley) [reply README.md:22]: @huddlej +1 for keeping vanilla Conda for now and including parallel instructions with Mamba.

**Comment by huddlej** [comment]
That would be great! Do you want to mention the option to update from Bioconda in the [`check_for_new_version` function](https://github.com/nextstrain/cli/blob/789d4a9a131bd239d4299eb924023aee36bb3ed2/nextstrain/cli/util.py#L52) or are you ok with this as it is?

Before we merge this PR, I need to update [the new update instructions](https://docs.nextstrain.org/en/latest/guides/install/local-installation.html#update-an-existing-installation). If we include the CLI in this PR, updating the environment will look like:

```
conda activate nextstrain

# Update Nextstrain dependencies (mafft, etc.), Augur, and Nextstrain CLI.
conda update --all

# Update Auspice.
npm update --global auspice
```

File: nextstrain.yml
Comment (fanninpm) [inline nextstrain.yml:15]: Have you tried https://anaconda.org/conda-forge/awscli?

Comment (huddlej) [reply nextstrain.yml:15]: I just tried this today, but when I created a new environment with mamba, an old version of Snakemake (5.5.*) got installed instead of the latest version (although [we currently require Snakemake >=5.4.0](https://bioconda.github.io/recipes/augur/README.html) for Augur, we really prefer the latest).

When I tried to install with conda instead, the environment took much much longer to solve. When I added Snakemake to this conda environment to override Augur's requirements and set the min version to 5.26.*, the conda solver took a long time to solve the environment, but mamba quickly solved it and installed Snakemake 6.0.2 and awscli 1.19.20.

I think we eventually want to source all of our dependencies from conda channels, but for this PR I'd like to stick with awscli as a pip package.

**Comment by tsibley** [comment]
> That would be great! Do you want to mention the option to update from Bioconda in the [`check_for_new_version` function](https://github.com/nextstrain/cli/blob/789d4a9a131bd239d4299eb924023aee36bb3ed2/nextstrain/cli/util.py#L52) or are you ok with this as it is?

I do want to update that code to suggest the right thing when running under Conda, but I don't think it needs to block this PR?

**Comment by tsibley** [comment]
npm warns that the version resolved by `nodejs>=10` is in conflict with the [declared reqs of Auspice](https://github.com/nextstrain/auspice/blob/f4fa3fd499bfc89c5813213e21e9dec4d02c2041/package.json#L10):

```
npm WARN EBADENGINE Unsupported engine {
npm WARN EBADENGINE   package: 'auspice@2.23.0',
npm WARN EBADENGINE   required: { node: '>=10.8.0 <=13.14', npm: '>=6.2' },
npm WARN EBADENGINE   current: { node: 'v15.11.0', npm: '7.6.0' }
npm WARN EBADENGINE }
```

but @jameshadfield says that Auspice's reqs can be updated to include the newer Node.js version.

**Comment by huddlej** [comment]
And auspice still installs and works fine with the later node, so that's probably not blocking here either. I'm ok with merging this now, unless you want to resolve the Auspice issue first, @tsibley.

**Comment by tsibley** [comment]
There are plenty of other warnings from npm that get ignored, so I think fine to merge before updating Auspice.

I do think we should strive to make the npm install as clean and warnings-free as possible as they can be confusing and lead folks to ignore actual warnings/errors due to fatigue/noise.

**Comment by tsibley** [comment]
> > That would be great! Do you want to mention the option to update from Bioconda in the [`check_for_new_version` function](https://github.com/nextstrain/cli/blob/789d4a9a131bd239d4299eb924023aee36bb3ed2/nextstrain/cli/util.py#L52) or are you ok with this as it is?
> 
> I do want to update that code to suggest the right thing when running under Conda, but I don't think it needs to block this PR?

Created https://github.com/nextstrain/cli/issues/114 to track this.

**Pull request merged**
